### PR TITLE
HYPR-432 Trace thread switches

### PR DESCRIPTION
**Pull request by jsvapiav**
Signed-off-by: Ivan Kuznetsov <jsvapiav@gmail.com>

### CODE REVIEW

**Comment by elmankku** [comment]
There are quite many checks failing in CI - those seem valid and related to the change, possibly fails on different configurations. Please fix.

File: include/api/debug.h [hunk 1]
```diff
@@ -77,7 +77,7 @@ static inline void debug_printUserState(void)
 
 static inline void debug_printTCB(tcb_t *tcb)
 {
-    printf("%40s\t", TCB_PTR_DEBUG_PTR(tcb)->tcbName);
+    printf("%lx %40s\t", (unsigned long int)tcb, TCB_PTR_DEBUG_PTR(tcb)->tcbName);
```
Comment (elmankku) [inline include/api/debug.h:80]: Is this still useful?

**Comment by jsvapiav** [comment]
> There are quite many checks failing in CI - those seem valid and related to the change, possibly fails on different configurations. Please fix.

**Comment by jsvapiav** [comment]
> There are quite many checks failing in CI - those seem valid and related to the change, possibly fails on different configurations. Please fix.

As far as I understand this commit will not be able to compile until this one will be merged:
`https://github.com/tiiuae/tii_sel4_build/pull/27`
So, I'm waiting when it will be merged and after going to recheck CI errors. That is a plan =)

File: include/api/debug.h (hunk 1 above)
Comment (jsvapiav) [reply include/api/debug.h:80]: I think this question mostly to @hlyytine .

**Comment by elmankku** [comment]
Okay. These are checks by upstream CI, which are mostly valid for us too - we want to avoid changes that break other configurations, or otherwise we're digging ourselves into a hole and makes upstreaming changes harder.

If the build requires an option to be set, then the incompatible parts should be flagged accordingly.

**Comment by jsvapiav** [comment]
> Okay. These are checks by upstream CI, which are mostly valid for us too - we want to avoid changes that break other configurations, or otherwise we're digging ourselves into a hole and makes upstreaming changes harder.
> 
> If the build requires an option to be set, then the incompatible parts should be flagged accordingly.

You are totally right. So, fixed. But still there are some CI errors which I can not easily understand. Could you please have a look on them?

**Comment by elmankku** [comment]
These preprocess checks are related to proofs, see: https://github.com/seL4/ci-actions/tree/master/preprocess#preprocess-check-for-sel4

We can ignore them.

File: include/api/debug.h (hunk 1 above)
Comment (hlyytine) [reply include/api/debug.h:80]: Yes, with this we can dump pointer to TCB and it's name -- later on we can decode the thread names from the pointers, that is, we can have human readable names in kernelshark etc output.

**Comment by hlyytine** [comment]
CI problems were because some of our repos tracked upstream and some were left behind. Should work now.

File: include/api/debug.h (hunk 1 above)
Comment (hlyytine) [reply include/api/debug.h:80]: Hmm I just realized it's quite a feat to collect something that seL4 prints to UART to use with kernelshark... So in this context this is not needed and could be removed. But something to keep in mind that the human readable name is found from that struct.

Comment (jsvapiav) [reply include/api/debug.h:80]: I think that actually it is better to keep it for now. And probably remove it when we adopt `human readable names` in kernelshark.

**Comment by jsvapiav** [comment]
> CI problems were because some of our repos tracked upstream and some were left behind. Should work now.

Yeah, you did fix everything except one last case (Process AARCH64). Great work!

**Comment by hlyytine** [comment]
In the near future we shall do that seL4_Yield() from guest Linux so that our seL4 kernel will be unmodified. Hence not fixing that remaining error.

**Pull request merged**